### PR TITLE
Fix split conviction vote

### PR DESCRIPTION
**Pull request by gui1117**
Fix https://github.com/galacticcouncil/HydraDX-ui/issues/1480

AFAICT hydration runtime handles split conviction:

https://github.com/galacticcouncil/hydration-node/blob/769c33d63d24356791c2f0e276350ebdc2914005/pallets/staking/src/integrations/democracy.rs#L41-L54

So I implemented the same logic here.

Also I am not good at typescript so please review carefully.

I still have to test it manually, but I don't really know how to use the testnet.

Also is there integration tests?

Hydration address:
7N5ixUUAHfyynZFDfXzB9FLnAAKgynWb95jBdujCLu1e4wLc

### CODE REVIEW

File: src/api/democracy.ts [hunk 1]
```diff
@@ -19,6 +20,26 @@ const CONVICTIONS_BLOCKS: { [key: string]: number } = {
...
+  }
+}
+
+const voteAmount = (vote?: PalletDemocracyVoteAccountVote): BN => {
+  if (vote?.isSplit) {
+    return vote.asSplit.toBigNumber() + vote.asSplit.nay.toBigNumber()
+  } else if (vote?.isStandard) {
+    return vote.asStandard.balance.toBigNumber()
+  } else {
+    return BN_0
```
Comment (thiolliere) [inline src/api/democracy.ts:39]: I honestly don't know if it better to return undefined or a default value here in ReactJS

File: src/api/democracy.ts [hunk 2]
```diff
@@ -19,6 +20,26 @@ const CONVICTIONS_BLOCKS: { [key: string]: number } = {
   locked6x: 1612800,
 }
 
+const voteConviction = (vote?: PalletDemocracyVoteAccountVote): String => {
+  if (vote?.isSplit) {
+    return 'None'
+  } else if (vote?.isStandard) {
+    return vote.asStandard?.vote.conviction.toString()
+  } else {
+    return 'None'
```
Comment (thiolliere) [inline src/api/democracy.ts:29]: I honestly don't know if it better to return undefined or a default value here in ReactJS

File: src/api/democracy.ts [hunk 3]
```diff
@@ -19,6 +20,26 @@ const CONVICTIONS_BLOCKS: { [key: string]: number } = {
...
+  }
```
Comment (vkulinich-cl) [inline src/api/democracy.ts:30]:
```suggestion
if (vote?.isStandard) {
    return vote.asStandard?.vote.conviction.toString()
  } else {
    return undefined
  }
```

File: src/api/democracy.ts [hunk 4]
```diff
@@ -19,6 +20,26 @@ const CONVICTIONS_BLOCKS: { [key: string]: number } = {
   locked6x: 1612800,
 }
 
+const voteConviction = (vote?: PalletDemocracyVoteAccountVote): String => {
```
Comment (vkulinich-cl) [inline src/api/democracy.ts:23]:
```suggestion
const voteConviction = (vote?: PalletDemocracyVoteAccountVote) => {
```

File: src/api/democracy.ts [hunk 5]
```diff
@@ -19,6 +20,26 @@ const CONVICTIONS_BLOCKS: { [key: string]: number } = {
...
+  } else if (vote?.isStandard) {
+    return vote.asStandard?.vote.conviction.toString()
+  } else {
+    return 'None'
+  }
+}
+
+const voteAmount = (vote?: PalletDemocracyVoteAccountVote): BN => {
+  if (vote?.isSplit) {
+    return vote.asSplit.toBigNumber() + vote.asSplit.nay.toBigNumber()
```
Comment (vkulinich-cl) [inline src/api/democracy.ts:35]:
```suggestion
     return vote.asSplit.aye.toBigNumber().plus(vote.asSplit.nay.toBigNumber())
```

File: src/api/democracy.ts (hunk 3 above)
Comment (thiolliere) [reply src/api/democracy.ts:30]: what about when the vote is split? the conviction should be defined at: `None`:

```typescript
    interface PalletDemocracyConviction extends Enum {
        readonly isNone: boolean;
        readonly isLocked1x: boolean;
        readonly isLocked2x: boolean;
        readonly isLocked3x: boolean;
        readonly isLocked4x: boolean;
        readonly isLocked5x: boolean;
        readonly isLocked6x: boolean;
        readonly type: 'None' | 'Locked1x' | 'Locked2x' | 'Locked3x' | 'Locked4x' | 'Locked5x' | 'Locked6x';
    }
 ```

In the conviction in the runtime the conviction for split vote is default:
https://github.com/galacticcouncil/hydration-node/blob/769c33d63d24356791c2f0e276350ebdc2914005/pallets/staking/src/integrations/democracy.rs#L53

Comment (vkulinich-cl) [reply src/api/democracy.ts:30]: it doesn't matter since None doesn't have any value and you won't get any action points

Comment (vkulinich-cl) [reply src/api/democracy.ts:30]: but yeah, let's leave None

Comment (vkulinich-cl) [reply src/api/democracy.ts:30]: @thiolliere will you apply that suggestion? I changed it to 'None'

**Comment by thiolliere** [comment]
Yes, applied, also the PR should be editable by maintainers, feel free to change it,
And considering you did most of the work with testing, I don't need tip

**Review by vkulinich-cl — APPROVED** [review]

**Review by jak-pan — APPROVED** [review]

File: src/api/democracy.ts [hunk 6]
```diff
@@ -131,12 +150,10 @@ export const getAccountUnlockedVotes =
...
 
     const votes = votesRaw.asDirect.votes.map(([id, dataRaw]) => {
-      const test = dataRaw.asStandard
-
       return {
         id: id.toString(),
-        balance: test.balance.toBigNumber(),
-        conviction: test.vote.conviction.toString(),
+        balance: voteAmount(dataRaw),
+        conviction: voteConviction(dataRaw)
```
Comment (jak-pan) [inline src/api/democracy.ts:156]:
```suggestion
        conviction: voteConviction(dataRaw),
```

**Comment by jak-pan** [comment]
IDK what is happening here but the actions got stuck with the new GH upgrade... Can't no longer approve workflow run

**Comment by vkulinich-cl** [comment]
Merged in the https://github.com/galacticcouncil/hydration-ui/pull/1525

**Pull request closed without merging**